internal/web/api/api.go

```diff
@@ -107,3 +115,83 @@ func (a *AlloyAPI) getClusteringPeersHandler() http.HandlerFunc {
 		_, _ = w.Write(bb)
 	}
 }
+
+func (a *AlloyAPI) liveDebugging() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		componentID := livedebugging.ComponentID(vars["id"])
+		requestedComponent := component.ParseID(vars["id"])
+
+		component, err := a.alloy.GetComponent(requestedComponent, component.InfoOptions{})
+		if err != nil {
+			http.NotFound(w, r)
+			return
+		}
+
+		if !a.debugCallbackManager.IsRegistered(livedebugging.ComponentName(component.ComponentName)) {
+			http.Error(w, fmt.Sprintf("Live debugging is not supported for the component \"%s\"", component.ComponentName), http.StatusInternalServerError)
+			return
+		}
+
+		// Buffer of 1000 entries to handle load spikes and prevent this functionality from eating up too much memory.
+		// TODO: in the future we may want to make this value configurable to handle heavy load
+		dataCh := make(chan string, 1000)
+		ctx := r.Context()
+
+		sampleProb := setSampleProb(w, r.URL.Query().Get("sampleProb"))
+
+		id := livedebugging.CallbackID(uuid.New().String())
+
+		a.debugCallbackManager.AddCallback(id, componentID, func(data string) {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				if sampleProb < 1 && rand.Float64() > sampleProb {
+					return
+				}
+				// Avoid blocking the channel when the channel is full
+				select {
+				case dataCh <- data:
+				default:
+				}
```

Comment:
As a future improvement, we might want to report back to the user when data is dropped (because of a full buffer) so they know that not everything is showing when the sample rate is 100%.

Reply:
Created an issue to track it: https://github.com/grafana/alloy/issues/1010